packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -52,24 +52,28 @@ export class NotificationsResourceHandler extends Construct {
   public readonly functionArn: string;
 
   /**
-   * The role of the handler's lambda function.
+   * The default policy of role of the handler's lambda function.
    */
-  public readonly role: iam.IRole;
+  private readonly policy: iam.Policy;
```

Comment:
This is breaking. A user could be using this property which now would become `private`